src/common/rearrangectrl.cpp

```diff
@@ -70,7 +70,8 @@ bool wxRearrangeList::Create(wxWindow *parent,
     }
 
     // do create the real control
-    m_order.reserve(count);
+    m_order = order;
```

Comment:
Could you please explain why was this changed?

Reply:
Shortly after the reservation of memory in the `m_order` member at line 73, there is a loop that attempts to reorder the member. However attempting to assign to the member in the loop in this fashion: `m_order[n] = order[n]` caused the program to throw an exception. I believe this is because `reserve()` only sets the capacity and not the size of the array, so when we try to access an element in the array there are none there.

Reply:
Yes, you're absolutely right, the current code is invalid and doesn't work at all if `order` argument is not empty. Apparently this got broken in 7e1b64a2eb6  -- @a-wi do you remember why was this particular change done? I understand the other changes in this commit, but not what was wrong with `m_order = order`.

In any case, we need to either revert this part of that commit or replace `reserve()` with `resize()`. The current version is just broken, while the new version doesn't make much sense: why assign the whole array first and then assign it again in the loop, the second assignment is useless (and removing it would be effectively the same as reverting part of 7e1b64a2eb6).

So I won't apply this part of the changes now, but we do need to fix it.

Reply:
No problem @vadz  . I have implemented your suggested change of using `resize()` rather than `reserve()`.